File: src/main/kotlin/com/full/stack/assignment/f1/data/cache/mapper/ConstructorMapper.kt
package com.full.stack.assignment.f1.data.cache.mapper

import com.full.stack.assignment.f1.data.cache.entity.ConstructorEntity
import com.full.stack.assignment.f1.model.Constructor
import org.springframework.stereotype.Component

@Component
class ConstructorMapper : Mapper<Constructor, ConstructorEntity> {
    override fun toDomain(entity: ConstructorEntity): Constructor {
        return Constructor(
            id = entity.id,
            name = entity.name,
            nationality = entity.nationality,
        )
    }

    override fun toEntity(domain: Constructor): ConstructorEntity {
        return ConstructorEntity(
            id = domain.id,
            name = domain.name,
            nationality = domain.nationality,
        )
    }
}

File: frontend/lib/data/remote/mappers/constructor_mapper.dart
import 'package:f1_api_client/api.dart';
import '../../../domain/models/constructor/constructor.dart';
import '../../../helpers/mapper/mapper.dart';

class ConstructorMapper extends Mapper<ConstructorDTO, Constructor, void> {
  @override
  Constructor map(ConstructorDTO param, {void metadata}) {
    return Constructor(
      constructorId: param.constructorId,
      name: param.name,
      nationality: param.nationality,
    );
  }
}
